Content.Server/Chemistry/Metabolism/MovespeedModifierMetabolism.cs

```diff
@@ -0,0 +1,88 @@
+        /// How much of the reagent should be metabolized each sec.
+        /// </summary>
+        [DataField("rate")]
+        public ReagentUnit MetabolismRate { get; set; } = ReagentUnit.New(1);
+
+        /// <summary>
+        /// How much the entities' walk speed is multiplied by.
+        /// </summary>
+        [DataField("walkSpeedModifier")]
+        public float WalkSpeedModifier { get; set; } = 1;
+
+        /// <summary>
+        /// How much the entities' run speed is multiplied by.
+        /// </summary>
+        [DataField("sprintSpeedModifier")]
+        public float SprintSpeedModifier { get; set; } = 1;
+
+        /// <summary>
+        /// how long the modifier persist after the final unit of reagent is metabolised,
+        /// should really be less than however long it takes for a metabolism tick(1 second).
+        /// </summary>
+        [DataField("statusLifetime")]
+        public int StatusLifetime = 3000;
+
+        /// <summary>
+        /// Remove reagent at set rate, changes the movespeed modifiers and adds a MovespeedModifierMetabolismComponent if not already there.
+        /// </summary>
+        /// <param name="solutionEntity"></param>
+        /// <param name="reagentId"></param>
+        /// <param name="tickTime"></param>
+        /// <returns></returns>
+        ReagentUnit IMetabolizable.Metabolize(IEntity solutionEntity, string reagentId, float tickTime)
+        {
+            if (solutionEntity.TryGetComponent(out MovementSpeedModifierComponent? movement))
+            {
+                solutionEntity.EnsureComponent(out MovespeedModifierMetabolismComponent status);
+
+                if (status.WalkSpeedModifier != WalkSpeedModifier)
+                {
+                    status.WalkSpeedModifier = WalkSpeedModifier;
+                }
+                if (status.SprintSpeedModifier != SprintSpeedModifier)
+                {
+                    status.SprintSpeedModifier = SprintSpeedModifier;
+                }
+                if (status.EffectTime != StatusLifetime * MetabolismRate.Int())
+                {
+                    status.EffectTime = StatusLifetime * MetabolismRate.Int();
+                }
+
+                ResetTimer(status);
+
+                //If any of the modifers aren't synced to the movement modifier component, then refresh them, otherwise don't
+                //Also I don't know if this is a good way to do a NAND gate in c#
+                if (!(status.WalkSpeedModifier.Equals(movement.WalkSpeedModifier) & status.SprintSpeedModifier.Equals(movement.SprintSpeedModifier)))
+                    movement?.RefreshMovementSpeedModifiers();
+
+            }
+            return MetabolismRate;
+        }
+        public void ResetTimer(MovespeedModifierMetabolismComponent status)
+        {
+            status.ModifierTimer = (_gametiming.CurTime, _gametiming.CurTime.Add(TimeSpan.FromMilliseconds(status.EffectTime)));
+            status.Dirty();
+        }
```

Comment:
1. These != are pointless as they'll never trigger.
2. EffectTime should really be incremental and not just "ping rounded up a bit" as 3s is just a dummy value to make sure the client receives it in time. After it's increased the timer should be dirtied to the client.
3. You should compare the status walk + sprint modifiers and _then_ work out whether it needs refreshing. You update it in advance so the Equals is pointless and it's annoying to read the code to read (also don't use a single &).